proposals/0024-repair-and-restart.md

```diff
@@ -0,0 +1,302 @@
+---
+simd: '0024'
+title: Optimistic cluster restart automation
+authors:
+  - Wen Xu (Solana Labs)
+category: Standard
+type: Core
+status: Draft
+created: 2023-04-07
+feature: (fill in with feature tracking issues once accepted)
+---
+
+## Summary
+
+During a cluster restart following an outage, use gossip to exchange local
+status and automatically reach consensus on the block to restart from. Proceed
+to restart if validators in the restart can reach agreement, or print debug
+information and halt otherwise.
+
+## New Terminology
+
+* "cluster restart": When there is an outage such that the whole cluster
+stalls, human may need to restart most of the validators with a sane state so
+that the cluster can continue to function. This is different from sporadic
+single validator restart which does not impact the cluster. See
+[cluster restart](https://docs.solana.com/running-validator/restart-cluster)
+for details.
+
+* "optimistically confirmed block": a block which gets the votes from the
+majority of the validators in a cluster (> 2/3 stake). Our algorithm tries to
+guarantee that an optimistically confirmed will never be rolled back. When we 
+are performing cluster restart, we normally start from the highest 
+optimistically confirmed block, but it's also okay to start from a child of the 
+highest optimistically confirmed block as long as consensus can be reached.
+
+* "silent repair phase": In the new repair and restart plan, the validators in
+restart will first spend some time to exchange information, repair missing
+blocks, and finally reach consensus. The validators only continue normal block
+production and voting after consensus is reached. We call this preparation
+phase where block production and voting are paused the silent repair phase.
+
+* "ephemeral shred version": right now we update `shred_version` during a
+cluster restart, it is used to verify received shreds and filter Gossip peers. 
+In the new repair and restart plan, we introduce a new temporary shred version 
```

Comment:
Another "new repair and restart plan" ref that probably should be renamed :)

Reply:
Fixed, I believe this is the last one.